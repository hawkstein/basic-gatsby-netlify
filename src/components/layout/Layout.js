import React from "react"
import PropTypes from "prop-types"
import { useStaticQuery, graphql } from "gatsby"

import Header from "./Header"
import "../../styles/reset.scss"
import "./layout.scss"

export const PureLayout = ({ children, data }) => {
  return (
    <>
      <div className="layout">
        <Header siteTitle={data.site.siteMetadata.title} />
        <main>{children}</main>
        <footer>
          All Rights Reserved Â© {data.site.siteMetadata.author}{" "}
          {new Date().getFullYear()}
        </footer>
      </div>
    </>
  )
}

const Layout = ({ children }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
          author
        }
      }
    }
  `)

  return <PureLayout data={data}>{children}</PureLayout>
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
