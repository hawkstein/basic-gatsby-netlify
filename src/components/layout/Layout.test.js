import React from "react"
import renderer from "react-test-renderer"
import { PureLayout } from "./Layout"
describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <PureLayout
          data={{
            site: {
              siteMetadata: {
                title: "Basic Gatsby Site",
                author: "liamhawks",
              },
            },
          }}
        >
          <p>Content</p>
        </PureLayout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
