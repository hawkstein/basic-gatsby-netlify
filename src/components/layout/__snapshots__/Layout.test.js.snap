// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="layout"
>
  <header>
    <span>
      <a
        href="/"
      >
        Basic Gatsby Site
      </a>
    </span>
  </header>
  <main>
    <p>
      Content
    </p>
  </main>
  <footer>
    Â© 
    liamhawks
     
    2020
  </footer>
</div>
`;
